pages/lib/template_graphs.py

```diff
@@ -844,6 +846,15 @@ def barchart(df, var, time_filter_info, data_filter_info, normalize, si_ip):
     return fig
 
 
+def time_filtering(df, start_time, end_time, time_col, target_col):
```

Comment:
all functions should have type annotation and a docstring, please add them both to all the function you have created.